network/gossip/libp2p/peerManager_test.go

```diff
@@ -114,108 +115,128 @@ func (ts *PeerManagerTestSuite) TestUpdatePeers() {
 
 		pm.updatePeers()
 
-		connector.AssertNumberOfCalls(ts.T(), "ConnectPeers", 4)
-		connector.AssertNumberOfCalls(ts.T(), "DisconnectPeers", 2)
+		connector.AssertNumberOfCalls(suite.T(), "ConnectPeers", 4)
+		connector.AssertNumberOfCalls(suite.T(), "DisconnectPeers", 2)
 	})
 }
 
-// TestPeriodicPeerUpdate tests that the peermanager runs periodically
-func (ts *PeerManagerTestSuite) TestPeriodicPeerUpdate() {
+// TestPeriodicPeerUpdate tests that the peer manager runs periodically
+func (suite *PeerManagerTestSuite) TestPeriodicPeerUpdate() {
 	currentIDs := unittest.IdentityListFixture(10)
 	idProvider := func() (flow.IdentityList, error) {
 		return currentIDs, nil
 	}
 
 	connector := new(mock.Connector)
-	connector.On("ConnectPeers", ts.ctx, testifymock.Anything).Return(nil)
-	connector.On("DisconnectPeers", ts.ctx, testifymock.Anything).Return(nil)
-	pm := NewPeerManager(ts.ctx, ts.log, idProvider, connector)
+	wg := &sync.WaitGroup{} // keeps track of number of calls on `ConnectPeers`
+	count, times := 0, 2    // we expect it to be called twice at least
+	wg.Add(times)
+	connector.On("ConnectPeers", suite.ctx, testifymock.Anything).Run(func(args testifymock.Arguments) {
+		if count < times {
+			count++
+			wg.Done()
```

Comment:
To avoid negative value on `WaitGroup`.

Reply:
From your comment below:
> Due to the asynchrony among assert.Eventually and peer update goroutine...

Will there be multiple goroutines concurrently calling `ConnectPeers`? If so I think we need to avoid concurrent writes to `count` in the `Run` function.

Reply:
Fixed in https://github.com/onflow/flow-go/pull/99/commits/cc4fad911c9fd1da51006d27b3da39934159869e.